### PR TITLE
docs: minor improvements for local deployment guide

### PR DESCRIPTION
**Pull request by SdgJlbl**
Minor improvements for the local deployment guide. 

The first commit proposes various improvements following remarks from first-time users. 

The second commit removes the option to install the frontend directly with `npm` in favour of a deployment using `Docker`.

### CODE REVIEW

File: docs/source/contributing/local-deployment.rst [hunk 1]
```diff
@@ -169,32 +163,28 @@ Launching
...
+        cd substra-frontend
+        docker build -f docker/substra-frontend/Dockerfile --target dev -t substra-frontend .
+        docker run -it --rm -p 3000:3000 -e API_URL=http://substra-backend.org-1.com -v ${PWD}/src:/workspace/src substra-frontend
 
-     | with ``DOCKER_FRONTEND_CONTAINER_NAME`` the name of the frontend container that will be used for the rest of the operations.
+  You can access the frontend at http://substra-frontend.org-1.com:3000/. The dev credentials are:
 
-  * In both case, you can access the frontend at http://substra-frontend.org-1.com:3000/.
+    * login: org-1
+    * password: p@sswr0d44
```
Comment (RomainGoussault) [inline docs/source/contributing/local-deployment.rst:187]: the @ in the password creates an mailto hyperlink :/. Do we have a way to remove the hyperlink?
image

Comment (RomainGoussault) [reply docs/source/contributing/local-deployment.rst:187]: Another suggestion to make the login/pwd stand out (I haven't tested this)
```suggestion
    * login: ``org-1``
    * password: ``p@sswr0d44``
```

**Review by RomainGoussault — COMMENTED** [review]
Thanks for the PR!

File: docs/source/contributing/local-deployment.rst (hunk 1 above)
Comment (SdgJlbl) [reply docs/source/contributing/local-deployment.rst:187]: well spotted 👍

**Review by RomainGoussault — APPROVED** [review]
Thanks for the PR.

Another approval from a more experienced engineer would be great.

File: docs/source/contributing/local-deployment.rst [hunk 2]
```diff
@@ -206,16 +196,8 @@ Stopping
...
 
-* Stop the frontend: This action depends on which option you chose during the launch:
-
-  a. Local server: Stop the process running the local server (usually using *Control+C* or *Command+C* on macOS)
-  b. Docker:
-
-     .. code-block:: bash
-
-        docker stop DOCKER_FRONTEND_CONTAINER_NAME
+* Stop the frontend: Stop the process running the local server in Docker (usually using *Control+C* or *Command+C* on macOS)
```
Comment (oleobal) [inline docs/source/contributing/local-deployment.rst:199]: On my laptop ⌘+C doesn't work for this purpose and I've always used Ctrl+C -- I thought it was pretty elegant how Ctrl+C always send SIGINT and ⌘+C always copies (compared to the X11 solution).

File: docs/source/contributing/local-deployment.rst [hunk 3]
```diff
@@ -169,32 +163,28 @@ Launching
 
       skaffold run -p dev,arm64
 
+.. tip::
+   If you need to re-run `skaffold run` for whatever reason, don't forget to use `skaffold delete` to reset the state beforehand (or reset your environment by running the `k3-create.sh` script again).
```
Comment (oleobal) [inline docs/source/contributing/local-deployment.rst:167]: Possibly the second part is a bit superfluous, there are many ways to reset the environment so giving one might be enough.

**Review by oleobal — APPROVED** [review]

**Review by oleobal — APPROVED** [review]
I left comments but I think it's a good improvement as-is already.

File: docs/source/contributing/local-deployment.rst (hunk 2 above)
Comment (SdgJlbl) [reply docs/source/contributing/local-deployment.rst:199]: yes, that's indeed the case, I've copied-pasted existing instructions without reading them 🤓

File: docs/source/contributing/local-deployment.rst (hunk 3 above)
Comment (SdgJlbl) [reply docs/source/contributing/local-deployment.rst:167]: Sometimes the `skaffold delete` ends with errors, and remembering to use the `k3-create` script again has helped me as a total k8s newbie to set things straight again.

**Pull request merged**